client/src/app/components/audit-template/audit-template-view/audit-template-view.component.ts

```diff
@@ -67,6 +129,83 @@ export abstract class AuditTemplateViewComponent implements OnInit {
     }
   }
 
+  timeZoneChange(event: { value: { utc: string[]; }; }): void {
+    this.timeZoneUTC = event.value.utc[0];
+  }
+
+  recurrenceExpand(): void {
+    this.repeatEvery = '1';
+    this.isRecurrenceChosen = true;
+    this.panelOpenState = true;
+  }
+
+  recurrenceCollapsed(): void {
+    this.isRecurrenceChosen = false;
+    this.panelOpenState = false;
+    this.allDaysChecked = false;
+    this.allMonthsChecked = false;
+    this.repeatEvery = '1';
+    this.recurrenceDay.subCheckBox?.forEach((t) => (t.checked = false));
+    this.recurrenceMonth.subCheckBox?.forEach((t) => (t.checked = false));
+    // @ts-ignore
+    this.recurrenceDay.subCheckBox[0].checked = true;
+    // @ts-ignore
+    this.recurrenceMonth.subCheckBox[0].checked = true;
+    this.errorMessageCheckboxDay = ' ';
+    this.errorMessageCheckboxMonth = ' ';
+  }
+
+  updateAllCheckbox(type: string): void {
+    if (type === 'dayCheckbox') {
+      this.allDaysChecked =
+        this.recurrenceDay.subCheckBox != null &&
+        this.recurrenceDay.subCheckBox.every((t) => t.checked);
+      this.errorMessageCheckboxDay = ' ';
+    } else if (type === 'monthCheckbox') {
+      this.allMonthsChecked =
+        this.recurrenceMonth.subCheckBox != null &&
+        this.recurrenceMonth.subCheckBox.every((t) => t.checked);
+      this.errorMessageCheckboxMonth = ' ';
+    }
+  }
```

Comment:
I didn't check to see if this would work, with the HTML, but I was wondering, could you split this into 2 functions and avoid having to pass in a string?

Reply:
updateAllCheckbox has now been split into two function named: updateCheckboxDay and updateCheckboxMonth.